File: College/Cplus-Book-Code/Catchup/Sort/bubbleSort.cpp
/*
 *  Created on: Feb 19, 2018
 *      Author: Master Ward
 */
#include <iostream>
#include <fstream>
#include <vector>
using namespace std;

void BubbleSort(int[], int);

int bubblesort() {
	/* Populating a vector */
	ifstream in;
	in.open("randInt.txt");
	vector<int> vc; // Temporary reading in vector
	vc.begin();
	int x;
	while (in >> x) {
		vc.push_back(x);
	}
	in.close();
	/* Start of program */
	int ar[vc.size()];
	copy(vc.begin(), vc.end(), ar);
	BubbleSort(ar, vc.size());
	for (unsigned int i = 0; i < 100; i++) {
		cout << ar[i] << " ";
	}
	return 0;
}
/* Sorts an array Ascendingly */
void BubbleSort(int array[], int size) {
	bool swap;
	int temp;
	do {
		swap = false;
		for (int i = 0; i < size - 1; i++) {
			if (array[i] > array[i + 1]) { // Sorting ascending
//			if(array[i] < array[i+1]) { // Sorts Descending
				temp = array[i];
				array[i] = array[i + 1];
				array[i + 1] = temp;
				swap = true;
			}
		}
	} while (swap);
}

File: High School/HS-Useful/Useful/src/Bubble_Sort.java

/**
 *
 * @author Master
 */
public class Bubble_Sort {

    public static void main(String[] args) {
        int n, c, d, swap;
        int a[] = {10, 2, 9, 56, 3, 56, 2, 4, 6, 4, 2, 3, 42, 2, 11, 1, 2, 3, 4, 54, 6, 7, 8, 9, 10, 0, 0, 11, 1};
        n = a.length;
        int array[] = new int[n];
        for (c = 0; c < n; c++) {
            array[c] = a[c];
        }
        for (c = 0; c < (n - 1); c++) {
            for (d = 0; d < n - c - 1; d++) {
                if (array[d] > array[d + 1]) {
                    swap = array[d];
                    array[d] = array[d + 1];
                    array[d + 1] = swap;
                }
            }
        }
        for (c = 0; c < n; c++) {
            System.out.print(array[c] + " ");
        }
    }
}
